python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -542,15 +552,8 @@ def _update_reservation(self):
 
         self._total_shared = remaining
 
-    def can_submit_new_task(self, op: PhysicalOperator) -> bool:
-        if op not in self._op_budgets:
-            return True
-        budget = self._op_budgets[op]
-        res = op.incremental_resource_usage().satisfies_limit(budget)
-        return res
-
-    def get_budget(self, op: PhysicalOperator) -> ExecutionResources:
-        return self._op_budgets[op]
+    def get_budget(self, op: PhysicalOperator) -> Optional[ExecutionResources]:
+        return self._op_budgets.get(op, None)
```

Comment:
```suggestion
        return self._op_budgets.get(op)
```

Reply:
I intentionally changed this to return an optional. Because ineligible ops don't have budgets.

Reply:
Yeah, i understand that but what's the point of specifying default? It will return none anyways

Reply:
got it. somehow I thought it raises key error by default.